### PR TITLE
Lw2688 assignment 6

### PR DESCRIPTION
**Pull request by lw2688**
Achieved the result that validated loss was much reduced to 1.5368 compared to 2.1038 in the first RNN model and validated loss is lower than training loss, which is a good fit avoid overfitting problem.

Other than the above, in this newly developed model, extra RNN layer was added to enable learn more complex language structure while the hidden neurons were reduced in order to avoid memory allocation error.

Sequences of 150 characters were adopted to trace language patterns further back.

The number of epochs were augmented to 60 intending to intensify the training process during which in this case the training loss and validation loss are consecutively reduced with validation smaller indicating avoidance of over-fitting problem.

Dropout probability was also altered from 0.5 to 0.3 to accelerate the training speed and minimizing the over-fitting.

Scale factors (temperature) were set respectively at 0.5 and 1.2 to get different prediction results which show that lower temperature produces more reasonable and conservative predictions.

### CODE REVIEW

**Comment by danilnagy** [comment]
Looks good! Great description of your methodology for tuning hyper-parameters.
